File: 908-MiddleOfTheLinkedList/908-MiddleOfTheLinkedList.cpp
// Last updated: 2026. 7. 16. 오전 11:26:20
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode* entry = head;
        int length = 0;
        while (entry) {
            ++length;
            entry = entry->next;
        }
        length /= 2;
        while (length--) {
            head = head->next;
        }
        return head;
    }
};

File: 908-MiddleOfTheLinkedList/908-MiddleOfTheLinkedList.java
// Last updated: 2026. 7. 16. 오전 11:26:17
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode tur = head;
        ListNode rab = head;
        while (rab != null && rab.next != null) {
            tur = tur.next;
            rab = rab.next.next;
        }
        return tur;
    }
}